api/test/media_integration.py

```diff
@@ -14,6 +14,21 @@ def search(fixture):
     assert fixture["result_count"] > 0
 
 
+def search_all_excluded(media_path, excluded_source):
+    response = requests.get(
+        f"{API_URL}/v1/{media_path}?q=test&excluded_source={','.join(excluded_source)}"
+    )
+    data = json.loads(response.text)
+    assert data["result_count"] == 0
+
+
+def search_source_and_excluded(media_path):
+    response = requests.get(
+        f"{API_URL}/v1/{media_path}?q=test&source=x&excluded_source=x"
```

Comment:
nit: It might make sense to change this to have different values for the two parameters, to emphasize the fact that the expected error is with _any_ mixing of the two parameters, not just the obvious contradiction of including and excluding a provider.

```suggestion
        f"{API_URL}/v1/{media_path}?q=test&source=x&excluded_source=y"
```